docs/introduction-to-smart-contracts.rst

```diff
@@ -432,6 +434,12 @@ you should minimize what you store in persistent storage to what the contract ne
 Store data like derived calculations, caching, and aggregates outside of the contract.
 A contract can neither read nor write to any storage apart from its own.
 
+Similar to storage, there is another data area called **transient storage**, where the main difference
```

Comment:
Should we add "Transient Storage" to the title of the section?

Also below it talks about "memory" as "the second data area" - if we talked about storage and transient storage already, might make sense to refer to it as the third.

Reply:
Also: weird that this section doesn't mention calldata or returndata or code or anything... (especially ``calldata``, since it's properly exposed on the language level)

Reply:
I added "Transient Storage" to the title.
Regarding the lack of mention to calldata/returndata/code, I will add a small paragraph about them here...I guess should be simple enough

Reply:
``calldata`` would be enough (since the others are there on the EVM level, but not really explicitly in the language really) - and we can also postpone it, it's not really related to this PR itself

Reply:
But yeah, a small paragraph would definitely be enough - they're all read-only, so it'd be enough to mention that there's also a those read-only data locations for storing arguments and return values across call frames, and accessing code (maybe mentioning that immutables live in code) or so.

Reply:
I just added a little section about them.